### PR TITLE
feat: add tes runs

### PR DESCRIPTION
**Pull request by JaeAeich**
## Description
This PR adds `runs` component, which is a list view for task runs. The `app-author` will be able to configure it by providing :
- `baseURL` - The base URL for their WES instance or it defaults to proWES deployment.
- `pageSize` - The number of runs per page (default: 5)
- `fields` - The structural configuration of expanded view (look in the documentation for more details).
- `filter` - The component should have a `filter-by-state` feature. (suitable for WES instances that have yet not implemented it).
- `search` - The component should have a `search-by-prefix` feature. (suitable for WES instances that have yet not implemented it).
> Note: Filtering is not implemented as backend has no support for it atm.
```html
<-- Default -->
<ecc-client-lit-ga4gh-tes-runs />

<-- With config -->
<ecc-client-lit-ga4gh-tes-runs 
  .baseURl="https://myBaseUrl.com/v1"
  .pageSize=11
  .fields=${myFieldsArray}
  .filter=${myFilterValueTrueOrFalse}
  .search=${mySearchValueTrueOrFalse}
/>
```  

## Checklist

- [x] My code follows the [contributing guidelines][contributing] of this project.
- [x] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code in hard-to-understand areas.
- [x] I have updated the user-facing documentation to describe any new or changed behavior.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

### CODE REVIEW

**Comment by JaeAeich** [comment]
@git-anurag-hub I have removed filtering from here as well. Please review it, and let's merge it.

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+      );
+    }
+
+    return updatedFilters;
+  }
+
+  private async _fetchData(page = 1) {
+    try {
+      // If all the items have been cached, don't invoke API call
+      if (this.nextPageToken === null) return;
```
Comment (git-anurag-hub) [inline packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts:204]: This is not needed!

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+        this.pageSize,
+        this.nextPageToken[page - 1],
+        this.view,
+        this.namePrefix
+      );
+
+      const convertedData: ItemProp[] = [];
+      data.tasks?.forEach((task: { id: string; state: string }) => {
+        convertedData.push({
+          index: this.items.length + convertedData.length + 1,
```
Comment (git-anurag-hub) [inline packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts:216]: Put the index according to the items that have been fetched. See the logic for that here: https://github.com/elixir-cloud-aai/cloud-components/blob/main/packages/ecc-client-lit-ga4gh-wes/src/components/runs/runs.ts#L186

Also better to filter the items here!

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+            width="16"
+            height="16"
+            fill="currentColor"
+            class="bi bi-trash3"
+            viewBox="0 0 16 16"
+          >
+            <path
+              d="M6.5 1h3a.5.5 0 0 1 .5.5v1H6v-1a.5.5 0 0 1 .5-.5M11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3A1.5 1.5 0 0 0 5 1.5v1H2.506a.58.58 0 0 0-.01 0H1.5a.5.5 0 0 0 0 1h.538l.853 10.66A2 2 0 0 0 4.885 16h6.23a2 2 0 0 0 1.994-1.84l.853-10.66h.538a.5.5 0 0 0 0-1h-.995a.59.59 0 0 0-.01 0zm1.958 1-.846 10.58a1 1 0 0 1-.997.92h-6.23a1 1 0 0 1-.997-.92L3.042 3.5zm-7.487 1a.5.5 0 0 1 .528.47l.5 8.5a.5.5 0 0 1-.998.06L5 5.03a.5.5 0 0 1 .47-.53Zm5.058 0a.5.5 0 0 1 .47.53l-.5 8.5a.5.5 0 1 1-.998-.06l.5-8.5a.5.5 0 0 1 .528-.47ZM8 4.5a.5.5 0 0 1 .5.5v8.5a.5.5 0 0 1-1 0V5a.5.5 0 0 1 .5-.5"
+            />
+          </svg>
```
Comment (git-anurag-hub) [inline packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts:307]: Slot for this is now deprecated!

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+      }
+    }
+  }
+
+  /* TODO: Add @ecc-utils-filter when backend implements filtering */
+  render() {
+    return html`
+      <ecc-utils-design-collection
+        id="collection"
+        .filters=${this.filters}
```
Comment (git-anurag-hub) [inline packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts:355]: Remove filter we will handle them later!

**Review by git-anurag-hub — CHANGES_REQUESTED** [review]

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts (hunk 2 above)
Comment (JaeAeich) [reply packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts:213]: oh yeah thanks! :slightly_smiling_face:

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts (hunk 1 above)
Comment (JaeAeich) [reply packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts]: done

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts (hunk 3 above)
Comment (JaeAeich) [reply packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts]: :man_facepalming: :+1:

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts (hunk 4 above)
Comment (JaeAeich) [reply packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts]: done

**Comment by JaeAeich** [comment]
@git-anurag-hub Should I have an image downloaded as a package like this one or will we use some hosted SVG. Wouldn't it be better to have a place where all our SVGs are then? Because this will cause inconsistency. Like even the one I am using rn doesn't fit.

**Comment by git-anurag-hub** [comment]
> @git-anurag-hub Should I have an image downloaded as a package like this one or will we use some hosted SVG. Wouldn't it be better to have a place where all our SVGs are then? Because this will cause inconsistency. Like even the one I am using rn doesn't fit.

Hey @JaeAeich, you can serve the icon right now from the package itself by putting it in /demo/assets directory. It will then be accessible from the `localhost:<port>/assets/<icon-name>` & can then pass the URL for the icon like `/assets/<icon-name>` so that the component can get the icon from the URL specified!

File: packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts [hunk 5]
```diff
@@ -282,6 +285,7 @@ export class TESRuns extends LitElement {
             key,
             name: "Delete",
             variant: "danger",
+            icon: "../../../assets/delete.svg",
```
Comment (git-anurag-hub) [inline packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts:288]: Icon will be served by the server & is not a relative path here!

**Review by git-anurag-hub — CHANGES_REQUESTED** [review]

**Review by git-anurag-hub — APPROVED** [review]

**Pull request merged**